awx/main/models/ha.py

```diff
@@ -499,6 +492,28 @@ def schedule_write_receptor_config(broadcast=True):
             write_receptor_config()  # just run locally
 
 
+# TODO: don't use the receiver post save, just call this at the moment when we need it
+# that way we don't call this multiple times unnecessarily
+@receiver(post_save, sender=ReceptorAddress)
+def receptor_address_saved(sender, instance, **kwargs):
+    from awx.main.signals import disable_activity_stream
+
+    address = instance
+
+    control_instances = set(Instance.objects.filter(node_type__in=[Instance.Types.CONTROL, Instance.Types.HYBRID]))
+    if address.peers_from_control_nodes:
+        address.peers_from.add(*control_instances)
+    else:
+        address.peers_from.remove(*control_instances)
```

Comment:
What about when a `ReceptorAddress` is saved, where that is the address for a control node?

We have specific rules are which control nodes are connected to which, and it is not bi-directional. I don't know what the value of `peers_from_control_nodes` will be for addresses in the control plane, and I don't know if it is relevant. But this would seem to fire for all saves of all receptor addresses, which would happen for the control nodes simply due to the fact that they have an associated receptor address, and it looks like that right away puts the links in the wrong state.

Reply:
I've since cleaned this up a bit. The logic is as follows:

if address has `peers_from_control_nodes` True, then have all control nodes peer to it
if address has `peers_from_control_nodes` False, then remove this peer from all control nodes.

I do a bit of set() comparisons to make sure the add/remove is necessary, because I want to minimize the amount of times we call `schedule_write_receptor_config`